crates/nu-utils/standard_library/tests.nu

```diff
@@ -37,20 +29,55 @@ def main [
     --command: string, # Test command to run. Default: all test command found in the files.
     --list, # Do not run any tests, just list them (dry run)
 ] {
-    let dry_run = ($list | default false)
-    for test_file in (collect-modules $path $module) {
-        let $module_name = ($test_file | path parse).stem
+    let tests = (
+        ls ($path | default $env.FILE_PWD | path join "test_*.nu")
+        | each {|row| {file: $row.name name: ($row.name | path parse | get stem)}}
+        | upsert test {|module|
+            nu -c $'use ($module.file) *; $nu.scope.commands | select name module_name | to nuon'
+            | from nuon
+            | where module_name == $module.name
+            | where ($it.name | str starts-with "test_")
+            | get name
+        }
+        | flatten
+        | rename file module name
+    )
 
-        log info $"Run tests in ($module_name)"
-        let tests = (collect-commands $test_file $module_name $command)
+    if $list {
+        return ($tests | select module name file)
+    }
 
-        for test_case in $tests {
-            log debug $"Run test ($module_name) ($test_case)"
-            if $dry_run {
-                continue
-            }
+    let tests_to_run = (if not ($command | is-empty) {
+        $tests | where name == $command
+    } else if not ($module | is-empty) {
```

Comment:
Can I set both module and command? It is important. For example more than one module can have the `test_output` command...

Reply:
Also I suggest not parsing the commands of a module if it is filtered out. There can be many modules and parsing all of them can be slow if I want to run only one module.

Reply:
i'll see what i can do :wink:

Reply:
it is a bit more work here and i really would like to fix the CI :thinking: 

could you tell me if that was working like this in your #8499 ?
if not i'd leave this for another PR :relieved: 

you tell me, in the meantime i'm gonne try to fix the CI :muscle:

Reply:
Yes, it worked in #8499. The `collect-modules` filtered the modules, _then_ the `collect-commands` filtered the commands. You leave it for another PR. CI integration has higher priority.

Reply:
ooooh i see :thinking: 

i'll see how much the CI takes and how much this becomes a mess :laughing: 

i've written this down not to forget it :relieved: